itest/lnd_channel_backup_test.go

```diff
@@ -1561,21 +1566,33 @@ func assertDLPExecuted(ht *lntest.HarnessTest,
 		// Dave should sweep his funds immediately, as they are not
 		// timelocked. We also expect Dave to sweep his anchor, if
 		// present.
-		ht.Miner.AssertNumTxsInMempool(expectedTxes)
+		if lntest.CommitTypeHasAnchors(commitType) {
+			ht.MineBlocksAndAssertNumTxes(1, 2)
+		} else {
+			ht.MineBlocksAndAssertNumTxes(1, 1)
+		}
 
-		// Mine the sweep tx.
-		ht.MineBlocksAndAssertNumTxes(1, expectedTxes)
+		blocksMined++
 
 		// Now Dave should consider the channel fully closed.
 		ht.AssertNumPendingForceClose(dave, 0)
 
+		// The above block will trigger Carol's sweeper to reconsider
+		// the anchor sweeping. Because we are now sweeping at the fee
+		// rate floor, the sweeper will consider this input has
+		// positive yield thus attempts the sweeping.
+		if lntest.CommitTypeHasAnchors(commitType) {
+			ht.MineBlocksAndAssertNumTxes(1, 1)
+			blocksMined++
+		}
+
 		// After Carol's output matures, she should also reclaim her
 		// funds.
 		//
 		// The commit sweep resolver publishes the sweep tx at
 		// defaultCSV-1 and we already mined one block after the
```

Comment:
Nit: I think the comment is not quite right, we did already mine more than 1 block?

Reply:
yeah fixed - think we should avoid testing multiple commitment types in a single test in the future, it's a bit too involved to maintain🤦🏻